api/v1alpha1/clusterconfig_types.go

```diff
@@ -370,6 +376,45 @@ func (NFD) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type CSIProviders struct {
+	// +optional
+	Providers []*CSIProvider `json:"providers,omitempty"`
+	// +optional
+	DefaultClassName string `json:"defualtClassName,omitempty"`
```

Comment:
I don't see how this get used, should we remove it for now and keep it simple and not set a default SC?

Reply:
We should leave it. I plan on adding this functionality as a part of this PR.

Reply:
implemented.